src/ansible_navigator/actions/settings.py

```diff
@@ -0,0 +1,201 @@
+"""The ``settings`` subcommand action."""
+
+import curses
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+from .._yaml import human_dump
+from ..action_base import ActionBase
+from ..action_defs import RunStdoutReturn
+from ..app_public import AppPublic
+from ..configuration_subsystem import transform_settings
+from ..steps import Step
+from ..ui_framework import CursesLinePart
+from ..ui_framework import CursesLines
+from ..ui_framework import Interaction
+from . import _actions as actions
+from . import run_action
+
+
+def filter_content_keys(obj: Dict[Any, Any]) -> Dict[Any, Any]:
+    """Filter out some keys when showing content.
+
+    :param obj: The object to be filtered
+    :return: The object with keys filtered out
+    """
+    return {k: v for k, v in obj.items() if not k.startswith("__")}
+
+
+def color_menu(colno: int, colname: str, entry: Dict[str, Any]) -> Tuple[int, int]:
+    # pylint: disable=unused-argument
+    """Color the menu.
+
+    :param colno: Column number
+    :param colname: Column name
+    :param entry: Column value
+    :return: Tuple int used to color the menu
+    """
+    if entry["is_default"]:
+        return 2, 0
+    return 3, 0
+
+
+def content_heading(obj: Any, screen_w: int) -> CursesLines:
+    """Create a heading for the setting entry showing.
+
+    :param obj: The content going to be shown
+    :param screen_w: The current screen width
+    :return: The heading
+    """
+    heading = []
+    string = obj["name"].replace("_", " ")
+    if obj["is_default"]:
+        string += f" (current/default: {obj['current_value']})"
+        color = 2
```

Comment:
I don't understand what this magic constant means but I suppose it's some sort of a color code. To improve readability, declare a module-level constant and use it here.

Wouldn't it be communicating the intent better if this read smth like
```suggestion
        color = COLOR_YELLOW
```
?

Reply:
The use of constants (or an enum)  here is a cool idea.  But....

all actions follow this pattern and should be changed at once and point to a set of shared definitions

Reply:
Well, I'm reviewing the patch I see in front of me. I can't talk about other places but this one is not possible to understand (I still don't know what these numbers are) and the improvements should start somewhere. Why not do this in the newly added code?

Reply:
If this gets merged, these can be used instead of the integers.  Having to define these as part of a PR introducing user facing functionality limited to one actions is,(IMO) a request for scope creep.

https://github.com/ansible/ansible-navigator/pull/985

Reply:
I suppose so. Although, I wouldn't be against declaring a few constants right in this module for the time being and refactoring that later. My only ask is not to use magic contextless constants here :man_shrugging:

Reply:
With that in mind, should we resolve this for this PR and handle it in #985 ?

Reply:
@alisonlhart we can resolve this once the constants from `curses` are used instead of the numbers.